### PR TITLE
fetch: Take spec file and an optional link as arguments

### PR DESCRIPTION
**Pull request by euanh**
All commands now take the spec file and an optional link as arguments.
This is necessary because the link can contain RPM macros which depend
on the spec file.

Fixes: #501

Signed-off-by: Euan Harris <euan.harris@citrix.com>

### CODE REVIEW

File: planex/Makefile.rules [hunk 1]
```diff
@@ -101,7 +101,7 @@ $(TOPDIR)/MANIFESTS/%.json:
 $(TOPDIR)/SOURCES/%:
 	@echo [FETCH] $@
 	$(AT) mkdir -p $(@D)
-	$(AT)$(FETCH) $(FETCH_FLAGS) $< $@
+	$(AT)$(FETCH) $(FETCH_FLAGS) $^ $@
```
Comment (mseri) [inline planex/Makefile.rules]: What does `$^` mean?

Comment (srowe) [reply planex/Makefile.rules]: All dependencies, rather than the leftmost dependency

Comment (euanh) [reply planex/Makefile.rules]: https://www.gnu.org/software/make/manual/make.html#Automatic-Variables

Comment (mseri) [reply planex/Makefile.rules]: Thanks!

**Pull request merged**